pkg/operator/operator.go

```diff
@@ -149,7 +149,7 @@ func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, erro
 		// return the updated resource. All client consumers will need audited to
 		// ensure they are tolerant of stale data (or we need a cache or client that
 		// makes stronger coherence guarantees).
-		NewClient: func(_ cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+		NewClient: func(config *rest.Config, options client.Options) (client.Client, error) {
```

Comment:
I think this change is causing the:
```
2023-05-23T20:45:45.130321593Z 2023-05-23T20:45:45.130Z ERROR   operator.main   cobra/command.go:920    error starting  {"error": "failed fetching infrastructure config: the cache is not started, can not read   objects"}
```
We try to query via the client before the cache is started, and it did work because we had `_ cache.Cache` to de-link the client from the cache, but I think now they are linked again. 

I'm struggling a bit to find how to accomplish separate cilent and cache with this new update.

Reply:
Figure it out, `options.cache = nil` override in `NewClient` makes client not use cache.